### DISCUSSION TITLE
Re-export `matchQuery` from utils module of the core package

### DISCUSSION THREAD

**Discussion by remolueoend** [TanStack/query#5064] [Ideas] [answered]
Hi everyone, I've got a small proposal I'd like to discuss: we would like to make use of the `matchQuery` function declared by the `utils` module in the `core` package.

#### Problem:
Since the entrypoints of the core package are limited using `exports` in `package.json`, this function can't be imported anymore by clients.

#### Rationale:
We currently employ a coarse-grained cache invalidation strategy, where we usually invalidate most cache keys. In certain situations, we need to exclude a small set of keys and their sub-keys from invalidation.

To implement our strategy, we make use of the `queryClient.invalidateQueries` API and provide a custom predicate to filter out certain cache keys. To do so, the function `matchQuery` has been proven to be very useful.

#### Solution:
The function `matchQuery` could be re-exported in `index.ts` of the `core` package.

We fully understand that this change increases the API surface of the package. We hope it is still worth a discussion. Thanks a lot to everyone for looking into it.

**Comment by TkDodo** [comment, marked as answer]
I think that's fine - PRs welcome 🚀

**Reply by remolueoend** [reply]
Thank you, I've opened a new PR: https://github.com/TanStack/query/pull/5070
